pipenv/utils.py

```diff
@@ -770,3 +770,15 @@ def find_requirements(max_depth=3):
                 if os.path.isfile(r):
                     return r
     raise RuntimeError('No requirements.txt found!')
+
+
+def remove_json_packages(pkg, bad_pkgs):
+    """Crawl pkg for bad packages in bad_pkgs to remove"""
+    # Check for dependencies to remove
+    pkgs = pkg.get('dependencies', False)
+    if pkgs:
+        pkgs = [p for p in pkgs if p['package_name'] not in bad_pkgs]
+        # Recursively crawl for bad package entries and remove them
+        pkgs = [remove_json_packages(p, bad_pkgs) for p in pkgs]
```

Comment:
FYI I considered the performance impact of iterating over a list twice.  I decided that it is probably not very significant, and opted to make this more legible.

I wasn't sure if we wanted to remove nested packages, so I designed for that and figured we could refactor if not.  But yes, if you wanted to prune nested dependency trees it would return what you wanted.  If there were no nested trees then each dependency would have no items in its 'dependencies' list and the `if pkgs` test would fail.

Academic point, but this code does what it set out to do.